File: backend/internal/model/metrics.go
package model

import "time"

// Metrics is the set of repository health metrics exposed by the dashboard API.
type Metrics struct {
	Repo        string    `json:"repo"`
	WindowDays  int       `json:"window_days"`
	GeneratedAt time.Time `json:"generated_at"`

	// PR metrics
	AvgTimeToAddressPRHours float64 `json:"avg_time_to_address_pr_hours"`
	AddressedPRCount        int     `json:"addressed_pr_count"`
	DisengagedPRCount       int     `json:"disengaged_pr_count"`
	DisengagedMultiplier    float64 `json:"disengaged_multiplier"`

	// Bug issue metrics (issues whose title is prefixed with "[BUG]")
	OpenBugIssueCount      int     `json:"open_bug_issue_count"`
	AvgTimeToCloseBugHours float64 `json:"avg_time_to_close_bug_hours"`
	ClosedBugCount         int     `json:"closed_bug_count"`
}

File: frontend/src/types/metrics.ts
export interface Metrics {
  repo: string
  window_days: number
  generated_at: string

  avg_time_to_address_pr_hours: number
  addressed_pr_count: number
  disengaged_pr_count: number
  disengaged_multiplier: number

  open_bug_issue_count: number
  avg_time_to_close_bug_hours: number
  closed_bug_count: number
}
